geospaas/nansat_ingestor/managers.py

```diff
@@ -68,68 +68,119 @@ def get_or_create(self,
-                warnings.warn('%s is not provided in Nansat metadata!' % name)
-            else:
-                try:
-                    value = json.loads(n_metadata[name])
-                except:
-                    warnings.warn('%s value of %s  metadata provided in Nansat is wrong!' %
-                                  (n_metadata[name], name))
-            options[name], _ = model.objects.get_or_create(value)
-
-        # Find coverage to set number of points in the geolocation
-        if len(n.vrt.dataset.GetGCPs()) > 0:
-            n.reproject_gcps()
-        geolocation = GeographicLocation.objects.get_or_create(
-            geometry=WKTReader().read(n.get_border_wkt(nPoints=n_points)))[0]
-
-        # create dataset
-        ds, created = Dataset.objects.get_or_create(
-            time_coverage_start=n.get_metadata('time_coverage_start'),
-            time_coverage_end=n.get_metadata('time_coverage_end'),
-            source=source,
-            geographic_location=geolocation,
-            **options)
+        entry_id = n_metadata['entry_id']
+        if len(Dataset.objects.filter(entry_id__icontains=entry_id)) >= 1:
+            qds = Dataset.objects.filter(
+                entry_id__icontains=entry_id)  # query of found dataset
+            assert len(qds) == 1
+            ds = qds.first()  # this dataset (which is the first and the only one) must be updated
+            # ds. Here is the place for setting the metadata from the nansat that comes from the localfile
+            if (n_metadata['platform'] is not None and
+                    n_metadata['instrument'] is not None):
+                platform, _ = Platform.objects.get_or_create(
+                    json.loads(n_metadata['platform']))
+                instrument, _ = Instrument.objects.get_or_create(
+                    json.loads(n_metadata['instrument']))
+                specs = n_metadata.get('specs', '')
+                source, _ = Source.objects.get_or_create(platform=platform,
+                                                         instrument=instrument,
+                                                         specs=specs)
+                ds.source = source  # update source
+
+            if 'entry_title' in n_metadata and n_metadata['entry_title'] is not None:
+                # update entry_title
+                ds.entry_title = n_metadata['entry_title']
+            if 'summary' in n_metadata and n_metadata['summary'] is not None:
+                ds.summary = n_metadata['summary']  # update summary
+            if 'time_coverage_start' in n_metadata and n_metadata['time_coverage_start'] is not None:
+                # update time_coverage_start
+                ds.time_coverage_start = n_metadata['time_coverage_start']
+            if 'time_coverage_end' in n_metadata and n_metadata['time_coverage_end'] is not None:
+                # update time_coverage_end
+                ds.time_coverage_end = n_metadata['time_coverage_end']
```

Comment:
This is repetition of exact same action 4 times. Respect DRY principle. Replace with a loop over metadata names.